R/dvhydrograph-render.R

```
@@ -3,199 +3,291 @@ dvhydrographPlot <- function(data) {
+    for (j in seq_len(length(refConfig))) {
+      plot_object <- do.call(names(refConfig[j]), append(list(object = plot_object), refConfig[[j]]))
     }
-    
-    plot_object <- ApplyApprovalBarStyles(plot_object, refData)
-    
-    plot_object <- rmDuplicateLegendItems(plot_object)
-    
-    plot_object <- plot_object %>% 
-      abline(v=seq(from=startDate, to=endDate, by="days"), lty=3, col="gray", where='first') %>%
-      abline(v=seq(from=startDate, to=endDate, by="weeks"), col="darkgray", lwd=1, where='first')
-    
-    # patch up top extent of y-axis
-    plot_object <- RescaleYTop(plot_object)
-    
-    return(plot_object)
   }
+  
+  plot_object <- ApplyApprovalBarStyles(plot_object, refData)
+  
+  plot_object <- rmDuplicateLegendItems(plot_object)
+  
+  plot_object <- plot_object %>% 
+    abline(v=seq(from=startDate, to=endDate, by="days"), lty=3, col="gray", where='first') %>%
+    abline(v=seq(from=startDate, to=endDate, by="weeks"), col="darkgray", lwd=1, where='first')
+  
+  # patch up top extent of y-axis
+  plot_object <- RescaleYTop(plot_object)
+  
+  return(plot_object)
 }
 
-getDVHydrographPlotConfig <- function(reportObject, info = NULL, ...){
+#' Get DV Hydrograph Plot Config
+#'
+#' @description Given an item to plot, fetch the associated
+#' plot feature(s) and their styles. 
+#' @param plotItem the item to fetch styles and plot features for
+#' @param ... any additional parameters to pass into the function
+getDVHydrographPlotConfig <- function(plotItem, ...){
   styles <- getDvHydrographStyles()
 
-  x <- reportObject[[1]]$time
-  y <- reportObject[[1]]$value
-  legend.name <- reportObject[[1]]$legend.name
+  x <- plotItem[[1]]$time
+  y <- plotItem[[1]]$value
+  legend.name <- plotItem[[1]]$legend.name
   args <- list(...)
   
-  styles <- switch(names(reportObject), 
-    stat1Timeseries = list(
-      lines = append(list(x=x, y=y, legend.name=legend.name), styles$stat1_lines)
+  styles <- switch(names(plotItem), 
```

I think we should add an exception after this switch statement if nothing is found. Would cover the case where someone is asking for a plotItem that's not supported/expected.

I will do that for UV Hydrographs.

---

yeah, that's just having something at the end without a name:
```r
nm <- "third"
switch(nm,
       first="stuff",
       second="more stuff",
       "my default")
[1] "my default"
```

---

Or you could throw an error or warning:

```r
nm <- "third"
switch(nm,
       first="stuff",
       second="more stuff",
       stop("not found"))
Error: not found

switch(nm,
       first="stuff",
       second="more stuff",
       warning("not found"))
Warning message:
not found 
```